ASC-993 linter not respecting valid python styling

mwbutcher opened the pull request:
Prior to this change, marks without parens were being
ignored.

This Commit adds logic in rules.py to diferentiate
between ast.Call and ast.Attribute and adds a couple
tests around this issue.

Commit:
ASC-993 linter not respecting valid python styling

Prior to this change, marks without parens were being
ignored.

This Commit adds logic in rules.py to diferentiate
between ast.Call and ast.Attribute and adds a couple
tests around this issue.

zreichert commented on tests/test_multi_value_mark.py:
```diff
@@ -43,6 +43,40 @@ def test_happy_path():
+    pass
+    """)
+    result = flake8dir.run_flake8(extra_args)
+    assert result.out_lines == []
```
I think this test should fail but I am not sure.  Since this mark implements a `value_regex` should it check to see if the values match the regex?

`pytest_mark1 = name=jira,value_regex=[a-zA-Z]*-\d*,allow_multiple_args=true`

I guess the question is 'Is an empty value acceptable, when you have defined a match or regex?'

zreichert reviewed (changes requested):
I think this behavior is not quite right.  In the example I commented on if we decorated a test with
`@pytest.mark.jira()`
It would fail our xml schema validation.

Commit:
ASC-993 linter not respecting valid python styling

As a side effect of this work, we discovered that there
was a case where a validator could be supplied for a mark
when no values were supplied. This should never be allowed.

This commit adds logic to fail in that case. This new check
has an additional implication. Since we drop any non-string
value, we'll end up in a situation where it seems like no
value has been supplied, so we'll fail for both having a
non-string value and also for not having a value.

Commit:
fixing lint errors

zreichert commented on flake8_pytest_mark/rules.py:
```diff
@@ -230,11 +235,18 @@ def _reduce_decorators_by_mark(decorators, mark):
+                    reduced.append(decorator)
+            except AttributeError:
+                pass
+        if isinstance(decorator, ast.Attribute):
```
Could this be `elif isinstance(decorator, ast.Attribute):`?

mwbutcher replied:
Yeah, those are mutually exclusive

Commit:
answering pr feedback
